packages/cli/core/src/commands/atomic/component.ts

```diff
@@ -0,0 +1,69 @@
+import {CLICommand} from '@coveo/cli-commons/command/cliCommand';
+import {Before} from '@coveo/cli-commons/decorators/before';
+import {Beta} from '@coveo/cli-commons/decorators/beta';
+import {UnknownError} from '@coveo/cli-commons/errors/unknownError';
+import {Flags} from '@oclif/core';
+import inquirer from 'inquirer';
+import {appendCmdIfWindows} from '../../lib/utils/os';
+import {spawnProcess} from '../../lib/utils/process';
+
+export default class AtomicInit extends CLICommand {
+  public static description =
+    'Scaffold a new custom component in your Coveo Atomic Component Library';
+  public static aliases = ['atomic:cmp'];
+
+  public static examples = [
+    '<%= config.bin %> <%= command.id %> --type=page myAwesomeComponent',
+    '<%= config.bin %> <%= command.id %> --type=result mySuperResultComponent',
+  ];
+
+  public static flags = {
+    type: Flags.string({
+      description: 'the kind of component to initialize',
+      options: ['page', 'result'],
```

Comment:
Hum, I'm not sure page is the correct name... 🤔 

In the doc, the classification that is used is `components` and `result template components`: https://docs.coveo.com/en/atomic/latest/reference/

Reply:
Could it be a boolean option instead ? `--resultComponent`

Reply:
I like the idea of a boolean option.

Reply:
I disagree here: if we want to add another kind of component/ component template, we would be stuck.

Reply:
Ah, the way I saw it, a result component is kind-of an extension of a page component: it behaves mostly the same, but it automatically receives a result as a parameter in some way.

I think it would be fine to have multiple boolean flags in the future if we add more components that are specific like that (e.g., `--quickviewComponent`). An error would be logged if we try to initialize with multiple of those flags.

However, I don't overall feel very strongly about using a boolean flag or an enum flag. An enum flag does have the advantage that it's more explicit, because if it's missing, we can warn the user that they must make a choice.

I don't know if I like the name "page" as an option, but I can't really think of a better name. Something like `default` implies that `result` may be the default, something like `generic` sounds like it refers to generic types, and something like `simple` implies that it's less powerful than `result`.